modules/core/src/main/scala/lucuma/odb/api/repo/ObservationRepo.scala

```diff
@@ -113,13 +117,15 @@ object ObservationRepo {
         }
 
 
-      override def insert(newObs: ObservationModel.Create): F[ObservationModel] = {
+      override def insert(newObs: ObservationModel.Create): F[ValidatedInput[ObservationModel]] = {
 
-        def construct(s: PlannedTimeSummaryModel): F[ObservationModel] =
+        def construct(s: PlannedTimeSummaryModel): F[ValidatedInput[ObservationModel]] =
           constructAndPublish { t =>
-            (tryNotFindObservation(t, newObs.observationId) *>
-             tryFindProgram(t, newObs.programId)            *>
-              newObs.withId(s)
+            (tryNotFindObservation(t, newObs.observationId)    *>
+             tryFindProgram(t, newObs.programId)               *>
+             newObs.asterismId.traverse(tryFindAsterism(t, _)) *>
+             newObs.targetId.traverse(tryFindTarget(t, _))     *>
```

Comment:
Checking the validity of the `asterismId` or `targetId` is unrelated and should have been done before now anyway.